src/components/modebar/index.js

```diff
@@ -12,42 +12,78 @@
 var Plotly = require('../../plotly');
 var d3 = require('d3');
 
+var Icons = require('../../../build/ploticon');
+var buttonsConfig = require('./buttons_config');
```

Comment:
What are the benefits in requiring this data structure instead of keeping the defaults in this file?